lib/util/query.js

```diff
@@ -148,22 +149,52 @@ export function getDefaultQuery (config) {
   return defaultQuery
 }
 
+/**
+ * Geocode utility for returning the first result for the provided place name text.
+ * @param  {String} text - text to search
+ * @param  {Object} geocoderConfig
+ * @return {Location}
+ */
+async function getFirstGeocodeResult (text, geocoderConfig) {
+  const geocoder = getGeocoder(geocoderConfig)
+  // Attempt to geocode search text and return first result if found.
+  // TODO: Import geocoder from @opentripplanner
+  return geocoder
+    .search({ text })
+    .then((result) => {
+      const firstResult = result.features && result.features[0]
+      if (firstResult) {
+        return geocoder.getLocationFromGeocodedFeature(firstResult)
+      }
+    })
+}
+
 /**
  * Create a otp query based on a the url params.
  *
  * @param  {Object} params An object representing the parsed querystring of url
  *    params.
  * @param config the config in the otp-rr store.
  */
-export function planParamsToQuery (params, config) {
+export async function planParamsToQuery (params, config) {
   const query = {}
   for (var key in params) {
     switch (key) {
       case 'fromPlace':
         query.from = parseLocationString(params.fromPlace)
+        // If a valid location was not found, but the place name text exists,
+        // attempt to geocode the name.
+        if (!query.from && params.fromPlace) {
+          query.from = await getFirstGeocodeResult(params.fromPlace, config.geocoder)
+        }
         break
       case 'toPlace':
         query.to = parseLocationString(params.toPlace)
+        // If a valid location was not found, but the place name text exists,
```

Comment:
Add a TODO comment to note potential for combining the code of these two cases.

Reply:
Fixed.